### PR TITLE
Get .NET Core compiling and tests running from Cake

### LINKED ISSUE #302: BadImageFormatException building solution with unmanaged projects
I have a solution with both managed and native projects. NUnit adapter seems to try to load a native DLL and fails whenever I rebuild the project. You can see the solution here: [Log4JDash](/MaxKot/Log4JDash/).

Error message (Log4JParserC.dll is an unmanaged library and Log4JDash.exe is an unmanaged application):

> ------ Discover test started ------
> NUnit Adapter 3.7.0.0: Test discovery starting
> Assembly contains no NUnit 3.0 tests: D:\develop\Log4JDash\src\Log4JDash.Web\bin\Log4JDash.Web.dll
> Assembly contains no NUnit 3.0 tests: D:\develop\Log4JDash\src\Log4JParserDashNet\bin\x86\Release\Log4JParserDashNet.exe
> Assembly contains no NUnit 3.0 tests: D:\develop\Log4JDash\src\Log4JParserNet\bin\x86\Release\Log4JParserNet.dll
> Error: Unable to get runner for this assembly. Check installation, including any extensions.
> BadImageFormatException: Format of the executable (.exe) or library (.dll) is invalid.
> Assembly not supported: D:\develop\Log4JDash\src\Release\Log4JDash.exe
> Error: Unable to get runner for this assembly. Check installation, including any extensions.
> BadImageFormatException: Format of the executable (.exe) or library (.dll) is invalid.
> Assembly not supported: D:\develop\Log4JDash\src\Release\Log4JParserC.dll
> NUnit Adapter 3.7.0.0: Test discovery complete
> ========== Discover test finished: 27 found (0:00:00,3920224) ==========

Visual Studio info:

> Microsoft Visual Studio Community 2015
> Version 14.0.25431.01 Update 3
> Microsoft .NET Framework
> Version 4.6.01055

Similar issue is [also reported](/nunit/nunit3-vs-adapter/issues/231#issuecomment-263929798) by @anttirt:

> I have the same issue as MaxKot; it's related to #248.
> 
> The issue is otherwise benign except that the `Error` log causes focus to switch to the `Tests` section in the output window after every build of the solution, which can be very annoying. :)
> 
> The NUnit runner _should_ somehow detect that a project is a native code project and skip it without emitting an error message, but I don't know if there's an API for that off the top of my head. The built in test discoverer for Visual Studio has the following to say:
> 
>```
> Test run will use DLL(s) built for framework Framework45 and platform X86. Following DLL(s) will not be part of run: 
> my_native_project.exe is built for Framework None and Platform X64.
>```
> 
> So perhaps skipping "assemblies" with `Framework = None` would do the trick?
> 
> I don't have time to experiment with this right now but may take a look at it later this week.

### LINKED ISSUE #395: Test do not run with .NET Core 2.0 installed
Since I installed .NET Core 2.0, the tests do not run from the command line,

```
.\build.ps1 -Configuration Release -Target Test
```

```
========================================
TestAdapterNetCore
========================================
Executing task: TestAdapterNetCore
The specified framework 'Microsoft.NETCore.App', version '1.0.5' was not found.
  - Check application dependencies and target a framework version installed at:
      C:\src\nunit\nunit3-vs-adapter\.dotnet\shared\Microsoft.NETCore.App
  - The following versions are installed:
      1.0.4
      1.1.1
  - Alternatively, install the framework version '1.0.5'.
```

### PR DESCRIPTION
**Pull request by rprouse**
I am not sure when this stopped working or exactly why. I thought it was because I installed .NET Core 2.0, but it seems to have been happening before that. Cake build was installing .NET Core 1.0.1 and adding it first in the path. This was causing the .NET Core app to fail to build.

To fix this, I first tried updating Cake to the latest version. This helped, but wasn't quite right because it installed .NET Core 1.0.4. I then switched the Cake bootstrap script to instead install .NET Core 1.1.4. This got everything compiling and the tests running.

This led to one .NET Core test failing. This was added after the tests broke in #369 to fix #302. No error or warning messages are reported by the adapter when attempting to load a native assembly using the .NET Core adapter, so I don't think #302 is an issue on .NET Core and so the test is invalid. I switched the test to only run full framework.

I also noticed that an old version of `NUnit3TestAdapterTests.csproj` got re-added to the directory, so I deleted it again. The project was renamed.

Fixes #395

### CODE REVIEW

File: build.ps1 [hunk 1]
```diff
@@ -31,12 +31,16 @@ Param(
     [string[]]$ScriptArgs
 )
 
-$CakeVersion = "0.19.3"
+$CakeVersion = "0.22.1"
 $DotNetChannel = "preview";
-$DotNetVersion = "1.0.1";
-$DotNetInstallerUri = "https://raw.githubusercontent.com/dotnet/cli/v1.0.1/scripts/obtain/dotnet-install.ps1";
+$DotNetVersion = "1.1.4";
```
Comment (rprouse) [inline build.ps1]: This file, `build.ps1` is a direct copy of the Cake 0.22.1 release version except for this one line that I updated from `1.0.4` to `1.1.4`. Everything else is as shipped.

File: build.cake [hunk 2]
```diff
@@ -5,7 +5,7 @@
 //////////////////////////////////////////////////////////////////////
 
 var target = Argument("target", "Default");
-var configuration = Argument("configuration", "Debug");
+var configuration = Argument("configuration", "Release");
```
Comment (rprouse) [inline build.cake]: Cake itself no longer passes in Release by default, so better to set it here to ensure that CI builds don't mistakenly switch to Debug. Charlie ran into this and we have been updating all repos this way.

**Comment by rprouse** [comment]
Woot, builds are green again! 🎉 😸

**Comment by OsirisTerje** [comment]
Looks good.

**Pull request merged**